examples/table_script/table_script.py

```diff
@@ -0,0 +1,157 @@
+"""
+Draft script to train a simple sklearn model 
+and output the results that could be used to create 
+a table similar to what is from the paper.
+
+"""
+
+import numpy as np
+import pandas as pd
+from pprint import pprint
+
+from rdkit import Chem, DataStructs
+from rdkit.Chem import AllChem
+
+from sklearn.svm import LinearSVR
+from sklearn.metrics import mean_squared_error, r2_score, mean_absolute_error
+
+from astartes import train_val_test_split
+
+
+# read in the data
+CSV_PATH = '../barrier_prediction_with_RDB7/ccsdtf12_dz.csv'
+df = pd.read_csv(CSV_PATH)
+df
+
+# helper function to featurize the data with 2048 morgan fingerprint
+# https://github.com/chemprop/chemprop/blob/master/chemprop/features/features_generators.py
+MORGAN_RADIUS = 2
+MORGAN_NUM_BITS = 2048
+def morgan_counts_features_generator(mol,
+                                     radius= MORGAN_RADIUS,
+                                     num_bits= MORGAN_NUM_BITS):
+    """
+    Generates a counts-based Morgan fingerprint for a molecule.
+    :param mol: A molecule (i.e., either a SMILES or an RDKit molecule).
+    :param radius: Morgan fingerprint radius.
+    :param num_bits: Number of bits in Morgan fingerprint.
+    :return: A 1D numpy array containing the counts-based Morgan fingerprint.
+    """
+    mol = Chem.MolFromSmiles(mol) if type(mol) == str else mol
+    features_vec = AllChem.GetHashedMorganFingerprint(mol, radius, nBits=num_bits)
+    features = np.zeros((1,))
+    DataStructs.ConvertToNumpyArray(features_vec, features)
+
+    return features
+
+# create X and y
+params = Chem.SmilesParserParams()
+params.removeHs = False
+
+X = np.zeros((len(df), 2048*2))
+for i, row in df.iterrows():
+    rsmi, psmi = row.rsmi, row.psmi
+    
+    rmol = Chem.MolFromSmiles(rsmi, params)
+    r_morgan = morgan_counts_features_generator(rmol)
+    
+    pmol = Chem.MolFromSmiles(psmi, params)
+    p_morgan = morgan_counts_features_generator(pmol)
+    
+    X[i, :] = np.concatenate((r_morgan,
+                              p_morgan - r_morgan),
+                             axis=0)
+
+y = df.dE0.values
+
+def produce_table(sklearn_model, X, y, samplers = ["random"], seed=0, hopts={},):
```

Comment:
This signature makes sense to me, and I like that we are specifying that it must be a sklearn model. Should help clarify things.

I will say that hopts should probably be a dict of dicts, with hyperparams for each of the samplers in `samplers`. the alternative would be to use `args`, and allow users to specify any number of sampler and optionally their hyperparams. Something like:
```python
produce_table(sklearn_model : sklearn.model, X, y=None, labels=None, *samplers)
...
for each sampler
    if next item in samplers is dict:
        call tvts with that dict and sampler
    else just call tvts
```

Reply:
I modified the signature a bit. Let me know if you have additional feedback. For now, I assume that `samplers` is a list of desired samplers and `sampler_hopts` is an optional dictionary of dictionaries. In theory, we could also omit the `samplers` list, but that would require the user to initialize a bunch of empty dictionaries if they just want to use the default parameters and my initial thought is that was marginally more cumbersome than just providing a list and only specifying hyperopts for the ones a user cares about.